src/codecs/imagequant/processor.ts

```diff
@@ -0,0 +1,43 @@
+import imagequant, { QuantizerModule } from '../../../codecs/imagequant/imagequant';
+import wasmUrl from '../../../codecs/imagequant/imagequant.wasm';
+import { QuantizeOptions } from './processor-meta';
+
+let emscriptenModule: Promise<QuantizerModule>;
+
+function initModule(): Promise<QuantizerModule> {
+  return new Promise((resolve) => {
+    const m = imagequant({
+      // Just to be safe, don’t automatically invoke any wasm functions
+      noInitialRun: false,
+      locateFile(url: string): string {
+        // Redirect the request for the wasm binary to whatever webpack gave us.
+        if (url.endsWith('.wasm')) return wasmUrl;
+        return url;
+      },
+      onRuntimeInitialized() {
+        // An Emscripten is a then-able that, for some reason, `then()`s itself,
+        // causing an infite loop when you wrap it in a real promise. Deleten the `then`
+        // prop solves this for now.
+        // See: https://github.com/kripken/emscripten/blob/incoming/src/postamble.js#L129
+        // TODO(surma@): File a bug with Emscripten on this.
+        delete (m as any).then;
```

Comment:
I know this isn't part of the PR, but could we change this so it doesn't require the delete?  Looking at the source, the thennable never fails, so we could just do:

```js
m.then(resolve)
```

Reply:
Also I bet we could extract the initialization into a helper method:

```js
// initWasmModule(imagequant, wasmUrl)
function initWasmModule(fn, wasmUrl) {
  return new Promise((resolve) => {
    const m = fn({
      // Just to be safe, don’t automatically invoke any wasm functions
      noInitialRun: false,
      locateFile(url: string): string {
        // Redirect the request for the wasm binary to whatever webpack gave us.
        if (url.endsWith('.wasm')) return wasmUrl;
        return url;
      },
      onRuntimeInitialized() {
        m.then(resolve);
      }
    });
  });
}
```

Reply:
> `m.then(resolve)`

Yeah, that seems less weird.

> Also I bet we could extract the initialization into a helper method

Agreed. Good point.